code/game/gamemodes/objective.dm

```diff
@@ -16,329 +16,290 @@ var/list/potential_theft_objectives=subtypesof(/datum/theft_objective) \
+		if(!ishuman(target.current))
+			return 1
+		if(iscarbon(target.current))
+			var/mob/living/carbon/C = target.current
+			if(C.handcuffed)
 				return 1
-			var/turf/T = get_turf(target.current)
-			// TODO: Tie into space manager
-			if(T && !(T.z in config.station_levels))			//If they leave the station they count as dead for this
-				return 2
-			return 0
-		return 1
 
-/datum/objective/mutiny/rp
-	find_target()
-		..()
-		if(target && target.current)
-			explanation_text = "Assassinate, capture or convert [target.current.real_name], the [target.assigned_role]."
-		else
-			explanation_text = "Free Objective"
-		return target
+		if(GAMEMODE_IS_REVOLUTION)
+			if(target in ticker.mode.head_revolutionaries)
+				return 1
 
+		// TODO: Tie into space manager
+		var/turf/T = get_turf(target.current)
+		if(T && !(T.z in config.station_levels))
+			return 1
+
+		return 0
+	return 1
 
-	find_target_by_role(role, role_type=0)
-		..(role, role_type)
-		if(target && target.current)
-			explanation_text = "Assassinate, capture or convert [target.current.real_name], the [!role_type ? target.assigned_role : target.special_role]."
-		else
-			explanation_text = "Free Objective"
-		return target
-
-	// less violent rev objectives
-	check_completion()
-		var/rval = 1
-		if(target && target.current)
-			//assume that only carbon mobs can become rev heads for now
-			if(target.current.stat == DEAD || target.current:handcuffed || !ishuman(target.current))
-				return 1
-			// Check if they're converted
-			if(istype(ticker.mode, /datum/game_mode/revolution))
-				if(target in ticker.mode:head_revolutionaries)
-					return 1
-			var/turf/T = get_turf(target.current)
-			// TODO: Tie into space manager
-			if(T && !(T.z in config.station_levels))			//If they leave the station they count as dead for this
-				rval = 2
-			return 0
-		return rval
 
 /datum/objective/maroon
 	martyr_compatible = 1
 
-	find_target()
-		..()
-		if(target && target.current)
-			explanation_text = "Prevent [target.current.real_name], the [target.assigned_role] from escaping alive."
-		else
-			explanation_text = "Free Objective"
-		return target
-
-	find_target_by_role(role, role_type=0)
-		..(role, role_type)
-		if(target && target.current)
-			explanation_text = "Prevent [target.current.real_name], the [!role_type ? target.assigned_role : target.special_role] from escaping alive."
-		else
-			explanation_text = "Free Objective"
-		return target
+/datum/objective/maroon/find_target()
+	..()
+	if(target && target.current)
+		explanation_text = "Prevent [target.current.real_name], the [target.assigned_role] from escaping alive."
+	else
+		explanation_text = "Free Objective"
+	return target
+
+/datum/objective/maroon/check_completion()
+	if(target && target.current)
+		if(target.current.stat == DEAD)
+			return 1
+		if(issilicon(target.current))
+			return 1
+		if(isbrain(target.current))
+			return 1
+		// TODO: Tie into space manager
+		if(target.current.z in config.admin_levels)

```

Comment:
Ack, sorry about the conflicts. Due to the way the assassinate check_completion is structured in this PR the changes I made in #5346 to the proc are irrelevant-- but this here undoes [this line](https://github.com/KasparoVy/Paradise/blob/6edee2b55a6feb580a882c96f6b10b1ca3a29f26/code/game/gamemodes/objective.dm#L180) and [this line](https://github.com/KasparoVy/Paradise/blob/6edee2b55a6feb580a882c96f6b10b1ca3a29f26/code/game/gamemodes/objective.dm#L181) from #5346